src/AppBundle/Entity/Post.php

```diff
@@ -133,13 +133,13 @@ public function setAuthorEmail($authorEmail)
     /**
      * Is the given User the author of this Post?
      *
-     * @param User $user
+     * @param User|string $user
```

Comment:
and what I'm saying is that the fix would be to stop passing a string to this method, not to accept strings and reject them as author